tamr_unify_client/client.py

```diff
@@ -89,7 +89,13 @@ def request(self, method: str, endpoint: str, **kwargs) -> requests.Response:
         Returns:
             HTTP response from the Tamr server
         """
-        url = urljoin(self.origin + self.base_path, endpoint)
+        # imperfect solution needed to allow passing absolute urls
+        if endpoint.count(":") == 1 and "/" not in endpoint:
+            # prefix endpoint with "./" to handle endpoints with a colon and no slashes
+            url = urljoin(self.origin + self.base_path, "./" + endpoint)
+        else:
+            # can handle with standard urljoin
+            url = urljoin(self.origin + self.base_path, endpoint)
```

Comment:
The only benefit of `urljoin` over just string concatenation is for handling absolute URLs. Maybe we should just replace this whole block with:

```suggestion
        if endpoint.startswith('/'):
            url = self.origin + endpoint
        else:
            url = self.origin + self.base_path + endpoint
```

Note: ^I forget if we have to add a `/` in between `self.base_path` and `endpoint`

Reply:
> The only benefit of `urljoin` over just string concatenation is for handling absolute URLs. Maybe we should just replace this whole block with:
> 
> Note: ^I forget if we have to add a `/` in between `self.base_path` and `endpoint`

It doesn't need the added `/`.  The only problem with this is that it no longer allows passing the full URI, e.g. 
`endpoint = http://localhost:9100/api/versioned/v1/instance:login` -> 
`url = http://localhost:9100/api/versioned/v1/http://localhost:9100/api/versioned/v1/instance:login`
We have tests around this that fail when I made the change.

Reply:
We could add a simple `if endpoint.startswith('http')` check as well

Reply:
That sounds pretty good.  My main concern was whether there are any cases where `http` and `https` are not the first part of the url, but I'm unfamiliar with this.

Reply:
To be more robust, we could use `urllib.parse` to inspect if there is a scheme https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

Reply:
This is where the API naming gets us into trouble, `urllib.parse.urlparse` will detect a scheme in the string "instance:login".

Reply:
Maybe we need to check for both a schema and a netloc?

Reply:
How do we know that a customer didn't actually implement a protocol called `instance` or `projects`?

We either need a mechanism to unambiguously differentiate between any arbitrary protocol and URL path segment OR we need to make assumptions about what the scheme (and other parts like netloc) look like.

Reply:
I made a change so that we look for any non-empty scheme and netloc, which I think does waht we need.  It uncovered another detail, which is that `urljoin` was also helping us sanitize urls occasionally. So if the endpoint was `projects/1/attributeConfigurations//1`, the `urljoin` call would remove the redundant second `/`.  One of our tests mistakenly caused this to happen, so I made a small change to that here too.
`